benchmark/benchmark.py

```diff
@@ -0,0 +1,30 @@
+import argparse
+from sys import argv
+
+from engine import *
+
+
+def _path(x):
+    return os.path.join(os.path.dirname(__file__), '../%s' % x)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--engine_type', type=str, required=True)
+    parser.add_argument('--gcp_credential_path', required=('DIALOGFLOW' in argv))
+    parser.add_argument('--gcp_project_id', required=('DIALOGFLOW' in argv))
+    parser.add_argument('--noise', required=True)
+
+    args = parser.parse_args()
+
+    engine = NLUEngine.create(NLUEngines[args.engine_type])
+    print('created %s engine' % str(engine))
+
+    if args.gcp_credential_path is not None:
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = args.gcp_credential_path
+
+    if args.gcp_project_id is not None:
+        engine._project_id = args.gcp_project_id

```

Comment:
accessing a protected member is not best practice. I think we should try to pass this to factory ".create" method somehow cleanly

Reply:
okay, I'll pass the project id into the .create method instead